.travis.yml

```diff
@@ -1,61 +1,79 @@
 ---
-sudo: required
 
 env:
-  - distribution: centos
+  - distribution: 'centos'
     version: 7
-    init: /usr/lib/systemd/systemd
-    run_opts: "--privileged --volume=/sys/fs/cgroup:/sys/fs/cgroup:ro"
-  - distribution: ubuntu
+    run_opts: '--privileged --volume="/sys/fs/cgroup:/sys/fs/cgroup:ro"'
+    init: '/usr/lib/systemd/systemd'
+  - distribution: 'ubuntu'
     version: 14.04
-    init: /sbin/init
-    run_opts: ""
-  - distribution: ubuntu
+    run_opts: ''
+    init: '/sbin/init'
+  - distribution: 'ubuntu'
     version: 16.04
-    init: /lib/systemd/systemd
-    run_opts: "--privileged --volume=/sys/fs/cgroup:/sys/fs/cgroup:ro"
+    run_opts: '--privileged --volume="/sys/fs/cgroup:/sys/fs/cgroup:ro"'
+    init: '/lib/systemd/systemd'
+
+language: 'python'
+
+python:
+  - '2.7'
 
 services:
-  - rabbitmq
-  - docker
+  - 'docker'
+  - 'rabbitmq'
+
+sudo: 'required'
+
 
 before_install:
-  # Pull container
-  - 'sudo docker pull ${distribution}:${version}'
-  # Customize container
-  - 'sudo docker build --rm=true --file=tests/Dockerfile.${distribution}-${version} --tag=${distribution}-${version}:ansible tests'
-  # Clone dependencies
-  - git clone --depth=1 https://github.com/openstack-ansible-galaxy/openstack-keystone.git ../openstack-keystone
+  - >-
+    docker build
+    --file=tests/Dockerfile.${distribution}-${version}
+    --tag=${distribution}-${version}:ansible
+    tests
+
+  - >-
+    docker run
+    --add-host="docker:$(/sbin/ip route | awk '/default/ { print $3 }')"
+    --detach
+    --env="ANSIBLE_FORCE_COLOR=True"
+    --env="ANSIBLE_HOST_KEY_CHECKING=False"
+    --env="ANSIBLE_LOG_PATH=/var/log/ansible.log"
+    --name="container"
+    --volume="${PWD}/tests:/etc/ansible:rw"
+    --volume="${PWD}:/etc/ansible/roles/openstack-glance:rw"
+    ${run_opts} ${distribution}-${version}:ansible "${init}"
+
+  - >-
+    docker exec container ansible-galaxy -c
+    install -r /etc/ansible/roles/openstack-glance/tests/requirements.yml
 
 install:
-  # Install requirements
-  - sudo apt-get update -qq
-  - sudo apt-get install python-pip -y
-  - sudo pip install ansible
-  - sudo ansible-galaxy install -r tests/requirements.yml
+  - 'pip install -r requirements-devel.txt'
 
   # TODO:  expose rabbitmq port to conatiners with dockerhost option
 script:
-  - container_id=$(mktemp)
-  # Run container in detached state
-  - >
-    sudo docker run --add-host=docker:`/sbin/ip route|awk '/default/ { print  $3}'` --detach
```

Comment:
I've added the `--add-host` part so the container can reach the rabbitmq service from the host, isn't this required ?